esp/bindings/http/platform/httptransport.cpp

```diff
@@ -1564,11 +1564,16 @@ void CHttpRequest::getBasicAuthorization(StringBuffer& userid, StringBuffer& pas
 
     StringBuffer uidpair;
     Utils::base64decode(authheader.length() - strlen("Basic "), authheader.str() + strlen("Basic "), uidpair);
+
+    //uidpair formatted as   [domain\]username:password    (domain optional)
     
+    //Look for optional domain name separator. Assumes username cannot contain '\'
     const char* pairstr = strchr(uidpair.str(), '\\');
+    if (pairstr && 0 == strchr(pairstr,':'))//Ensure domain name separator was not part of password. Assumes username cannot contain ':'
```

Comment:
This looks to me as though it will not handle the case where a \ and a password are both specified. I think the proper test is something like "pwstart = strchr(uidpair, ':'); if (pairstr && pwstatr && pwstart < pairstr) ..."